### PR TITLE
[WIP] Gstreamer based capture and ROS publisher

### PR DESCRIPTION
**Pull request by ar13pit**
Resolves tue-robotics/tue_robocup#666
Resolves tue-robotics/tue_robocup#697

### CODE REVIEW

**Comment by ar13pit** [comment]
Note to self:
Split commit `223132adc2b0fd0be1112b1f68a2bbc69fb81749` into two if edit is not possible. Keep the scripts and remove the data then rebase and force push.

File: launch/speech_recognition.launch
Comment (LoyVanBeek) [inline launch/speech_recognition.launch]: Please file issue for each `#TODO`

File: src/speech_recognition/hmi_server_kaldi_client.py
Comment (LoyVanBeek) [inline src/speech_recognition/hmi_server_kaldi_client.py:39]: If you want to use a sleep, maybe don't wait for a full second but maybe a 0.1s. OR, maybe even better; use a callback to let KaldiGstApp let HMIServerKaldiClient know there is a sentence ready.

**Review by LoyVanBeek — CHANGES_REQUESTED** [review]

File: launch/speech_recognition.launch
Comment (ar13pit) [reply launch/speech_recognition.launch]: I won't merge this branch without fixing this anyway. So it will be taken care of before the merge.

File: src/speech_recognition/hmi_server_kaldi_client.py
Comment (ar13pit) [reply src/speech_recognition/hmi_server_kaldi_client.py:39]: Also as I have not tested this yet, I am not sure about the reason of having a sleep command here. I'll do this tomorrow night as I need to test the killing of the gstreamer pipeline too.

Comment (LoyVanBeek) [reply src/speech_recognition/hmi_server_kaldi_client.py:39]: From what i gather from the code, there is a background process that sets `kaldi_app.sentence`. If that is not yet set, the background process is not yet done. You can check it without the sleep, but then would run a very busy loop until there is a sentence. Having s shorter sleep isn't nice, but a middle ground between this kind of polling and getting a callback (a notification)

**Pull request merged**
